core/domain/feedback_services.py

```diff
@@ -261,3 +262,27 @@ def enqueue_feedback_message_email_task(user_id):
     taskqueue_services.enqueue_task(
         feconf.FEEDBACK_MESSAGE_EMAIL_HANDLER_URL, {'user_id': user_id},
         feconf.DEFAULT_FEEDBACK_MESSAGE_EMAIL_COUNTDOWN_SECS)
+
+
+def send_feedback_message_email(exploration_id, thread_id, message_id):
+    """Creates"""
+    exploration_rights = rights_manager.get_exploration_rights(exploration_id)
+    feedback_message = {
+        'exploration_id': exploration_id,
+        'thread_id': thread_id,
+        'message_id': message_id
+    }
+
+    for owner in exploration_rights.owner_ids:
+        model = feedback_models.UnsentFeedbackEmailModel.get(
+            owner, strict=False)
+
+        if model:
+            model.feedback_messages.append(feedback_message)
+            model.put()
+
+        else:
+            model = feedback_models.UnsentFeedbackEmailModel(
```

Comment:
Well, you just don't want to get into a situation like this:
- Thread 1 gets the model from the datastore
- Thread 2 gets the model from the datastore
- Thread 1 writes the model back, with one item for feedback_messages
- Thread 2 writes the model back, with one item (a different one) for feedback_messages

Because the fourth operation comes after the third one, the information sent in the third one is lost.

Similarly, this also has issues:
- Thread 1 gets the model from the datastore
- Thread 2 gets the model from the datastore
- Thread 1 writes the model back, with one item for feedback_messages
- Thread 2 sends a bunch of emails from the model it retrieved, and then deletes the model

Again, the information sent in the third operation is lost.

So I don't think there's a cookie-cutter solution; you'll need to think through the cases. Probably the get-and-delete() also needs a transaction.